src/ray/gcs/gcs_client/service_based_accessor.cc

```diff
@@ -812,10 +813,11 @@ Status ServiceBasedNodeInfoAccessor::AsyncGetInternalConfig(
           callback(status,
                    std::unordered_map<std::string, std::string>(
                        reply.config().config().begin(), reply.config().config().end()));
-        } else {
+        } else if (!status.ok()) {
           RAY_LOG(ERROR) << "Failed to get internal config: " << status.message();
-          callback(status, boost::none);
         }
+        RAY_LOG(DEBUG) << "No internal config was stored.";
+        callback(status, boost::none);
```

Comment:
It will call callback twice?

Reply:
How about like this:       
        boost::optional<std::unordered_map<std::string, std::string>> config;
        if (status.ok()) {
          if (reply.has_config()) {
            RAY_LOG(DEBUG) << "Fetched internal config: " << reply.config().DebugString();
            config = std::unordered_map<std::string, std::string>(
                reply.config().config().begin(), reply.config().config().end());
          } else {
            RAY_LOG(DEBUG) << "No internal config was stored.";
          }
        } else {
          RAY_LOG(ERROR) << "Failed to get internal config: " << status.message();
        }
        callback(status, config);

Reply:
Good catch